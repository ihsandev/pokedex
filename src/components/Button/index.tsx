import { PokemonTypeColor } from "../../utils/_functions";
import styled from "styled-components";

const Button: any = styled.button`
  margin-top: ${({ top }: any) => top}px;
  margin-bottom: ${({ bottom }: any) => bottom}px;
  background-color: transparent;
  cursor: pointer;
  border: none;
  border: 1.5px solid ${({ color }: any) => PokemonTypeColor(color)};
  border-radius: 50px;
  padding: 10px 30px;
  outline: none;
  transition: all 0.2s ease;
  &:hover {
    background-color: ${({ color }: any) => PokemonTypeColor(color)};
    color: #fff;
  }
`;

export default Button;
