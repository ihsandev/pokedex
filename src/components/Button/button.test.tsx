import * as React from "react";
import { shallow } from "enzyme";
import Button from "./index";

test("render Button without props", () => {
  const wrapper = shallow(<Button>Test</Button>);
  expect(wrapper).toMatchSnapshot();
});

test("render Button with props", () => {
  const wrapper = shallow(
    <Button top={20} bottom={20} color="grass">
      Test
    </Button>
  );
  expect(wrapper).toMatchSnapshot();
});
