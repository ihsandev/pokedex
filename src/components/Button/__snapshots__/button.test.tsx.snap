// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Button with props 1`] = `
<StyledComponent
  bottom={20}
  color="grass"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  margin-top: ",
          [Function],
          "px;
  margin-bottom: ",
          [Function],
          "px;
  background-color: transparent;
  cursor: pointer;
  border: none;
  width: 100%;
  border: 1.5px solid ",
          [Function],
          ";
  border-radius: 50px;
  padding: 10px 30px;
  outline: none;
  transition: all 0.2s ease;
  &:hover {
    background-color: ",
          [Function],
          ";
    color: #fff;
  }
  ",
          "@media (min-width: 575.98px)",
          " {
    width: inherit;
  }
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  top={20}
>
  Test
</StyledComponent>
`;

exports[`render Button without props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  margin-top: ",
          [Function],
          "px;
  margin-bottom: ",
          [Function],
          "px;
  background-color: transparent;
  cursor: pointer;
  border: none;
  width: 100%;
  border: 1.5px solid ",
          [Function],
          ";
  border-radius: 50px;
  padding: 10px 30px;
  outline: none;
  transition: all 0.2s ease;
  &:hover {
    background-color: ",
          [Function],
          ";
    color: #fff;
  }
  ",
          "@media (min-width: 575.98px)",
          " {
    width: inherit;
  }
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Test
</StyledComponent>
`;
